pkg/tar/tar.go

```diff
@@ -192,9 +192,18 @@ func (t *stiTar) ExtractTarStream(dir string, reader io.Reader) error {
 					errorChannel <- err
 					break
 				}
+				if header.Mode&tar.TypeSymlink == tar.TypeSymlink {
+					if err := extractLink(dir, header, tarReader); err != nil {
+						glog.Errorf("Error extracting link %q: %v", header.Name, err)
+						errorChannel <- err
+						break
+					}
+					continue
+				}
 				if err := extractFile(dir, header, tarReader); err != nil {
-					glog.Errorf("Error extracting file %s: %v", header.Name, err)
+					glog.Errorf("Error extracting file %q: %v", header.Name, err)
```

Comment:
since you're changing this from %s to %q, could you do the same for the surrounding code?
https://github.com/openshift/source-to-image/pull/307/files#diff-6b4b95c3fb2a650ac8b9a0c5eabdb8b3R183
and
https://github.com/openshift/source-to-image/pull/307/files#diff-6b4b95c3fb2a650ac8b9a0c5eabdb8b3R191